File: 백준/Silver/11060. 점프 점프/점프 점프.cs
using System;

class Program
{
    static void Main() {
        int n = int.Parse(Console.ReadLine());
        string[] s = Console.ReadLine().Split(' ');
        int[] jump = new int[1001];
        int[] dp = new int[1001];
        for(int i=1;i<=n;i++)
        {
            jump[i] = int.Parse(s[i-1]);
            dp[i] = 987654321;
        }
        dp[1] = 0;
        for(int i = 1;i<=n;i++)
        {
            for(int j = 0;j<=jump[i];j++)
            {
                if((i+j)<=n) dp[i+j] = Math.Min(dp[i]+1,dp[i+j]);
            }
        }
        if(dp[n] == 987654321) Console.Write(-1);
        else Console.Write(dp[n]);
    }
}

File: 백준/Silver/11060. 점프 점프/점프 점프.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main(){
	int n, jump[1001], dp[1001]={};
	cin >> n;
	for(int i=1; i<=n; i++){
		cin >> jump[i];
		dp[i]=987654321;
	}
	dp[1]=0;
	for(int j=1; j<=n; j++){
		for(int k=0; k<=jump[j]; k++){
			if((j+k)<=n)
				dp[j+k]=min(dp[j]+1,dp[j+k]);
		}
	}
	if(dp[n]==987654321) cout << -1;
	else cout << dp[n];
}